benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubProtocolBenchmark.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Internal.Encoders;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+
+namespace Microsoft.AspNetCore.SignalR.Microbenchmarks
+{
+    [ParameterizedJobConfig(typeof(CoreConfig))]
+    public class HubProtocolBenchmark
+    {
+        private HubProtocolReaderWriter _hubProtocolReaderWriter;
+        private byte[] _binaryInput;
+        private TestBinder _binder;
+        private HubMessage _hubMessage;
+
+        [Params(Message.NoArguments, Message.FewArguments, Message.ManyArguments, Message.LargeArguments)]
+        public Message Input { get; set; }
+
+        [Params(Protocol.MsgPack, Protocol.Json)]
+        public Protocol HubProtocol { get; set; }
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            switch (HubProtocol)
+            {
+                case Protocol.MsgPack:
+                    _hubProtocolReaderWriter = new HubProtocolReaderWriter(new MessagePackHubProtocol(), new PassThroughEncoder());
+                    break;
+                case Protocol.Json:
+                    _hubProtocolReaderWriter = new HubProtocolReaderWriter(new JsonHubProtocol(), new PassThroughEncoder());
+                    break;
+            }
+
+            switch (Input)
+            {
+                case Message.NoArguments:
+                    _hubMessage = new InvocationMessage("123", true, "Target", null);
+                    break;
+                case Message.FewArguments:
+                    _hubMessage = new InvocationMessage("123", true, "Target", null, 1, "Foo", 2.0f);
+                    break;
+                case Message.ManyArguments:
+                    _hubMessage = new InvocationMessage("123", true, "Target", null, 1, "string", 2.0f, true, (byte)9, new byte[] { 5, 4, 3, 2, 1 }, 'c', 123456789101112L);
+                    break;
+                case Message.LargeArguments:
+                    _hubMessage = new InvocationMessage("123", true, "Target", null, new string('F', 1234), new byte[1000]);
```

Comment:
larger Larger 

# LARGER

But seriously, let's try 10 KB (10240) and see how it performs, if the benchmark is super slow, then scale it back. Also use a power of two, not a power of ten, jeez.

Reply:
Psh power of two is too clean, computers like those, we don't want to be nice to the computer